libdnf-cli/output/transaction_table.hpp

```diff
@@ -21,37 +21,57 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 #ifndef LIBDNF_CLI_OUTPUT_TRANSACTION_TABLE_HPP
 #define LIBDNF_CLI_OUTPUT_TRANSACTION_TABLE_HPP
 
+
+#include "libdnf-cli/utils/tty.hpp"
+#include "libdnf-cli/utils/units.hpp"
+
+#include <libdnf/rpm/nevra.hpp>
+
 #include <fmt/format.h>
 #include <libsmartcols/libsmartcols.h>
-#include <vector>
 
-#include "../../libdnf-cli/utils/units.hpp"
+#include <vector>
 
 
 namespace libdnf::cli::output {
 
 
-enum { COL_NEVRA, COL_REPO, COL_SIZE };
+enum { COL_NAME, COL_ARCH, COL_EVR, COL_REPO, COL_SIZE };
 
+
+/// Compare two packages by their Name, Arch and Epoch:Version-Release.
 template <class Package>
-void sort_pkgs_list(std::vector<Package> & list) {
-    sort(list.begin( ), list.end( ),
-            [ ]( const Package & lhs, const Package & rhs ) {
-        return lhs.get_full_nevra() < rhs.get_full_nevra();
-    });
-}
+bool cmp_package_naevr(const Package & a, const Package & b) {
+    // compare by name
+    bool result = a.get_name() < b.get_name();
+    if (result)
+        return result;
+
+    // compare by arch
+    result = a.get_arch() < b.get_arch();
+        return result;
+
+    // compare by evr using rpm's rpmvercmp()
+    result = libdnf::rpm::rpmvercmp(a.get_evr().c_str(), b.get_evr().c_str());
+    return result;
```

Comment:
and here too?

Reply:
The `cmp_package_naevr()` is I think wrong on multiple levels. It should not continue after finding out that names (or architectures later on) of packages are different. And than there is problem with `libdnf::rpm::rpmvercmp()`. This function returns integer -1 / 0 / 1 according to whether a < b, a==b or a > b. But than the number is converted to bool and both -1 and 1 are considered `true`.